r1cs/src/gadget/field.rs

```diff
@@ -57,11 +62,62 @@ impl<C: CircuitDriver> FieldAssignment<C> {
         z
     }
 
+    fn range_check(cs: &mut R1cs<C>, value: &Self, num_of_bits: u16) {
```

Comment:
Is the way to check binary decomposition correct?
I think we don't use inner_product but minus one check as following article says.
https://hackmd.io/@7dpNYqjKQGeYC7wMlPxHtQ/B1w_9nq2Y

Reply:
Fixed